R/mlr3learners_table.R

```diff
@@ -0,0 +1,130 @@
+# args:
+#  hide_cols `character()`: specify which, if any, columns to hide
+#  filter `list()`: named list of conditions to filter on, names correspond to column names
+#                   in table
+#  tibble `logical(1)`: if TRUE returns table as tibble otherwise data.table
+#
+# examples:
+#    list_mlr3learners(hide_cols = c("properties", "feature_types"),
+#                      filter = list(class = "surv", predict_types = "distr"))
+#    list_mlr3learners(tibble = TRUE)
+list_mlr3learners = function(hide_cols = NULL, filter = NULL, tibble = FALSE) {
+
+  dt = copy(learner_table)
+
+  class = mlr3_package = required_package = NULL # hacky fix to prevent NOTE for global binding
+
+  if (!is.null(filter)) {
+    if (!is.null(filter$class)) {
+      dt = subset(dt, class %in% filter$class)
+    }
+    if (!is.null(filter$mlr3_package)) {
+      dt = subset(dt, mlr3_package %in% filter$mlr3_package)
+    }
+    if (!is.null(filter$required_package)) {
+      dt = subset(dt, required_package %in% filter$required_package)
+    }
+    if (!is.null(filter$properties)) {
+      dt = subset(dt, mlr3misc::map_lgl(dt$properties,
+                                        function(.x) any(filter$properties %in% .x)))
+    }
+    if (!is.null(filter$feature_types)) {
+      dt = subset(dt, mlr3misc::map_lgl(dt$feature_types,
+                                        function(.x) any(filter$feature_types %in% .x)))
+    }
+    if (!is.null(filter$predict_types)) {
+      dt = subset(dt, mlr3misc::map_lgl(dt$predict_types,
+                                        function(.x) any(filter$predict_types %in% .x)))
+    }
+  }
+
+  if (!is.null(hide_cols)) {
+    dt = subset(dt, select = !(colnames(dt) %in% hide_cols))
+  }
+
+  if (tibble) {
+    return(tibble::tibble(dt))
+  } else {
+    return(dt)
+  }
+}
+
+
+# overloads lrn function to automatically detect and install learners from any packages in
+# mlr3verse. uses list_mlr3learners with filtering for the given key.
+# this should actually probably be implemented in mlr3misc::dictionary_sugar_get
+# however this would create a dependency loop unless the learners table also lives in mlr3misc.
+# a vectorised version of this for `lrns` follows naturally.
+#
+# the function filters the learner_table, searches to see if the required mlr3_package is installed
+# and if not uses usethis::ui_yeah to ask user to install, if yes then installed and learner loaded,
+# if not then errors
+#
+# args:
+#  .key `character(1)`: learner key
+#
+# examples:
+#
+# lrn("classif.ranger")
+#
+# unloadNamespace("mlr3learners.coxboost")
+# utils::remove.packages("mlr3learners.coxboost")
+# lrn("surv.coxboost")
+
+lrn = function(.key, ...) {
+  id = NULL # hacky fix to prevent NOTE for global binding
+  pkg = unlist(subset(list_mlr3learners(), id == .key)$mlr3_package)
+  inst = suppressWarnings(require(pkg, quietly = FALSE, character.only = TRUE))
+  if (!inst) {
+    ans = usethis::ui_yeah(
```

Comment:
{usethis} is a heavy dep and this should be pulled in.

If at all I'd like {cli} but this should discussed upstream and not here.
Otherwise use the message helpers in {mlr3misc}.

Reply:
I did it with `cat`, it's a fairly simple one so didn't need anything too exciting